third_party/nvfuser/csrc/lower_allocation.cpp

```diff
@@ -480,10 +481,18 @@ class AllocationInserter : public kir::ExprMutator {
             default_val == nullptr,
             "Welford should not have a default initialization value for predicate elimination.");
         init = expr->as<GroupedWelfordOp>()->getInitValOfOutput(out);
-      } else if (default_val != nullptr) {
+      } else {
         init = default_val;
       }
 
+      if (ir_utils::isCpAsyncOp(expr)) {
+        TORCH_CHECK(
+            init == nullptr || init->isZero(),
+            "cp.async initialized with non-zero is not supported");
+        // cp.async will automatically fill zero when out of bound
```

Comment:
Interesting. So, does that mean if the source buffer on global memory is accessed beyond its buffer limit, `cp.async` automatically returns zero?

Reply:
Yes. (assuming that your predicate is correct)